src/aui/auibook.cpp

```diff
@@ -3446,6 +3446,140 @@ bool wxAuiNotebook::InsertPage(size_t index, wxWindow *page,
     }
 }
 
+// Helper class to calculate the best size of a wxAuiNotebook
+class wxAuiLayoutObject
+{
+public:
+    wxAuiLayoutObject(const wxSize &size,
+                      const wxAuiPaneInfo &pInfo) : m_pInfo(pInfo)
+    {
+        m_size = size;
+        /* To speed up the sorting of the panes, the direction is mapped to a
+         * useful increasing value. This avoids complicated comparison of the
+         * enum values during the sort. The size calculation is done from the
+         * inner to the outermost direction. Therefore CENTER < LEFT/RIGHT <
+         * TOP/BOTTOM (It doesn't matter it LEFT or RIGHT is done first, as
+         * both extend the best size horizontally; the same applies for
+         * TOP/BOTTOM in vertical direction) */
+        switch (pInfo.dock_direction)
+        {
+            case wxAUI_DOCK_CENTER: m_dir = 0; break;
```

Comment:
Thanks to the comment above I now understand why do we do this, but could we please have some symbolic values instead of using manifest constants? Code like `m_dir < 3` below would be more clear if it used `DockDir_Horizontal` instead.

Reply:
Replaced the magic numbers and used `DockDir_Vertical` for the comparisons. Although it still needs some closer look to see, why `mergeHorizontal` is used for `< DockDir_Vertical` in one place and `>= DockDir_Verical` in the other.